Fluid/MemberAccessStrategyExtensions.cs

```diff
@@ -1,44 +1,44 @@
-﻿using System;
-using System.Collections.Concurrent;
+﻿using Fluid.Accessors;
+using System;
 using System.Collections.Generic;
 using System.Reflection;
 using System.Threading.Tasks;
-using Fluid.Accessors;
 
 namespace Fluid
 {
     public static class MemberAccessStrategyExtensions
     {
-        internal static ConcurrentDictionary<string, IMemberAccessor> _namedAccessors = new ConcurrentDictionary<string, IMemberAccessor>();
-        private static ConcurrentDictionary<Type, List<string>> _typeMembers = new ConcurrentDictionary<Type, List<string>>();
+        internal static Dictionary<string, IMemberAccessor> _namedAccessors = new Dictionary<string, IMemberAccessor>();
+        private static Dictionary<Type, List<string>> _typeMembers = new Dictionary<Type, List<string>>();
 
         private static List<string> GetAllMembers(Type type)
         {
-            return _typeMembers.GetOrAdd(type, t =>
+            if (!_typeMembers.TryGetValue(type, out var list))
             {
-                var list = new List<string>();
+
+                list = new List<string>();
 
                 foreach (var propertyInfo in type.GetTypeInfo().GetProperties(BindingFlags.Public | BindingFlags.Instance))
                 {
                     list.Add(propertyInfo.Name);
-                    _namedAccessors.TryAdd($"({type.FullName})-{propertyInfo.Name}", new MethodInfoAccessor(propertyInfo.GetGetMethod()));
+                    _namedAccessors[$"({type.FullName})-{propertyInfo.Name}"] = new MethodInfoAccessor(propertyInfo.GetGetMethod());
                 }
 
                 foreach (var fieldInfo in type.GetTypeInfo().GetFields(BindingFlags.Public | BindingFlags.Instance))
                 {
                     list.Add(fieldInfo.Name);
-                    _namedAccessors.TryAdd($"({type.FullName})-{fieldInfo.Name}", new DelegateAccessor((o, n) => fieldInfo.GetValue(o)));
+                    _namedAccessors[$"({type.FullName})-{fieldInfo.Name}"] = new DelegateAccessor((o, n) => fieldInfo.GetValue(o));
                 }
 
-                return list;
-            });
+                _typeMembers.Add(type, list);
+            }
+
+            return list;
         }
 
         public static IMemberAccessor GetNamedAccessor(Type type, string name)
         {
-            IMemberAccessor result = null;
-
-            return _namedAccessors.GetOrAdd($"({type.FullName})-{name}", key =>
+            if (!_namedAccessors.TryGetValue($"({type.FullName})-{name}", out var result))
```

Comment:
maybe benchmark the $ usage against string.Concat?

Reply:
Or might value tuple be good candidate for access too? requires benchmarking I guess.

Reply:
I think Value Type might be better, for allocations at least